app/models/timeline/date_adjustment.rb

```diff
@@ -28,7 +28,10 @@ def days_adjustment
     end
 
     def adjust_scheduled_date(step, days_adjustment)
-      if step.scheduled_start_date && days_adjustment != 0
+      # If step has a precedent step, do not change the date
+      if step.schedule_parent_id
+        false
+      elsif step.scheduled_start_date && days_adjustment != 0
```

Comment:
This looks correct but it needs test coverage. I'd suggest a spec that sets up four steps, one of them with a schedule_parent, attempts to adjust the dates of three of them (including the dependent one AND the one it depends on), checks that the operation still works and that all four steps still have the right start dates. (The dependent one should also change by virtue of its dependence, and this is an important thing to double check.)

Also, the proper test driven way to do this would be to comment out the new code, write the test, watch it blow up like it used to, then uncomment your new code and watch it pass. Few things are more satisfying in life than this, IMHO :)

There should already be a spec file for this class, so this would just be adding to it.

If you need to pair on this, let me know.